Add wizzard for guiding through purchase

Ekwav opened the pull request:
Given the plans to add even more payment options through google pay, I think the premium purchase process should be more guided.
This draft is currently missing updating the last step with the selected options

Ekwav committed:
Add wizzard for guiding through purchase

matthias-luger commented on components/Premium/PremiumPurchaseWizard/PremiumPurchaseWizard.tsx:
```diff
@@ -0,0 +1,294 @@
+                >
+                    <Card.Body className={styles.optionBody}>
+                        <div className={styles.optionIcon}>⭐</div>
+                        <h5>Starter Premium</h5>
```
The headlines of the cards are not readable:

<img width="797" height="1131" alt="Image" src="https://github.com/user-attachments/assets/7433a2e7-43dd-462c-ba69-8cf9a84edccf" />

matthias-luger commented on components/Premium/PremiumPurchaseWizard/PremiumPurchaseWizard.tsx:
```diff
@@ -0,0 +1,294 @@
+                    <p>You can use your earned CoflCoins to purchase {selectedTier === PremiumTier.STARTER ? 'Starter Premium' : selectedTier === PremiumTier.PREMIUM ? 'Premium' : 'Premium Plus'}.</p>
+                    <div className={styles.infoBox}>
+                        <CheckCircle className={styles.checkIcon} />
+                        <span>This step will take you directly to the purchase options.</span>
```
Also not readable:

<img width="771" height="504" alt="Image" src="https://github.com/user-attachments/assets/f99304ff-d377-4b4e-a959-078d05080cfc" />

matthias-luger reviewed (changes requested):
The steps should probably be seperate components

Ekwav committed:
Break up into components

Ekwav committed:
Fix colors and add two step checkout for coflcoins

Ekwav committed:
handle all steps

Ekwav committed:
polish

Ekwav committed:
its almost 6 years already of full data
and there were some auctions before so lets round it up
will take a while to get online anways

Ekwav committed:
adjust colors

Ekwav committed:
improve colors further

Ekwav committed:
clear slop from wiki

Ekwav committed:
note that price represents yearly

matthias-luger committed:
fix formating

matthias-luger committed:
fix spelling mistake

matthias-luger committed:
remove logging

matthias-luger committed:
Merge branch 'develop' into premium-wizzard
